spark/src/main/scala/io/delta/tables/execution/DeltaTableOperations.scala

```diff
@@ -112,6 +114,50 @@ trait DeltaTableOperations extends AnalysisHelper { self: DeltaTable =>
     toDataset(sparkSession, restore)
   }
 
+  protected def executeClone(
+      table: DeltaTableV2,
+      target: String,
+      replace: Boolean,
+      properties: Map[String, String],
+      versionAsOf: Option[Long] = None,
+      timestampAsOf: Option[String] = None): DeltaTable = withActiveSession(sparkSession) {
+    val (targetRelation, targetPath) = if (new Path(target).isAbsolute()) {
+      (UnresolvedRelation(TableIdentifier(target, Some("delta"))), Some(target))
+    } else {
+      val tableIdentifier = sparkSession
+        .sessionState
+        .sqlParser
+        .parseTableIdentifier(target)
+      (UnresolvedRelation(tableIdentifier), None)
+    }
+    val source = DataSourceV2Relation.create(table, None, table.getTableIdentifierIfExists.map(
+      id => Identifier.of(id.database.toArray, id.table)))
+
+    val maybeTimeTravel = if (timestampAsOf.isDefined || versionAsOf.isDefined) {
```

Comment:
```suggestion
    val timeTraveledSourceRelation = if (timestampAsOf.isDefined || versionAsOf.isDefined) {
```

Would be a cleaner name

Reply:
Name was taken from the clone sql parser https://github.com/delta-io/delta/blob/597950c5ff5cf0aa2b76a3d5fba9da58fdba332e/spark/src/main/scala/io/delta/sql/parser/DeltaSqlParser.scala#L302

Reply:
So perhaps can we call it `maybeTimeTravelSource`?